### PR TITLE
add DBContext Pooling

### PR DESCRIPTION
**Pull request by stefannikolei**
#253

### CODE REVIEW

**Comment by stefannikolei** [comment]
@brockallen i added the pooling. I also made it configurable.

**Comment by stefannikolei** [comment]
@brockallen do you have any feedback on this pr?

**Comment by brockallen** [comment]
Not yet sorry -- in the next week or two. Sorry for the delay. Thanks

File: src/EntityFramework.Storage/Configuration/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -36,24 +36,40 @@ public static class IdentityServerEntityFrameworkBuilderExtensions
...
+                {
+                    services.AddDbContext<TContext>(options.ResolveDbContextOptions);
+                }
             }
             else
             {
-                services.AddDbContext<TContext>(dbCtxBuilder =>
+                if (options.EnablePooling)
+                {
+                    services.AddDbContext<TContext>(dbCtxBuilder =>
```
Comment (Cephei) [inline src/EntityFramework.Storage/Configuration/ServiceCollectionExtensions.cs:61]: I'm new to ef connection pooling, but isn't this `if` logic mixed up with below?

Comment (stefannikolei) [reply src/EntityFramework.Storage/Configuration/ServiceCollectionExtensions.cs:66]: Good catch ty

**Comment by brockallen** [comment]
Hey -- finally had a look. Looks promising. 

I don't know what the default should be for pool size. 1024 feels high to me. Do you have a server running with 1K concurrent connections? Maybe we make it nullable and only use it if it's explicitly configured, and that way we get the EF defaults (thanks to @ardalis for that idea).

**Comment by stefannikolei** [comment]
> Hey -- finally had a look. Looks promising. 
> 
> 
> 
> I don't know what the default should be for pool size. 1024 feels high to me. Do you have a server running with 1K concurrent connections? 
> 
> 
> 
> Maybe we make it nullable and only use it if it's explicitly configured, and that way we get the EF defaults (thanks to @ardalis for that idea).
> 
> 
> 
> 

Yeah that sounds reasonable. Going to change it

**Comment by stefannikolei** [comment]
@brockallen i have changed the things you had requested

**Comment by brockallen** [comment]
FWIW, I did a very crude, non-scientific, non-profiled perf/load test all locally and I didn't see any significant improvements with the pooling enabled.

**Comment by brockallen** [comment]
Was there any more work you wanted to do on this? I do see several warnings about the new properties missing docs/comments, for example.

**Comment by brockallen** [comment]
I submitted a PR to your branch for some missing items.

**Comment by brockallen** [comment]
Let me know once you've merged it, and then I can merge this PR. I have other work I need to do, but it's blocked until this PR gets merged (just to avoid merge conflicts).

**Comment by stefannikolei** [comment]
@brockallen merged it

**Comment by brockallen** [comment]
Thanks!

**Comment by brockallen** [comment]
Hmm... something went wrong with the PR. I'll have to look into it.

**Pull request merged**